### PR TITLE
Expose table v2 sections

### PR DESCRIPTION
**Pull request by matheusps**
#### What is the purpose of this pull request?

Exposed table sections to increase extensibility. 

#### What problem is this solving?

[Running workspace](https://matheus--cosmetics1.myvtex.com/admin/collections)

#### How should this be manually tested?

clone the repo and `yarn && yarn start`

#### Types of changes

- [x] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/components/EXPERIMENTAL_Table/docs/Advanced.md [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+    </Table>
+  )
+}
+
+;<HeadExample />
+```
+
+##### Body
+
+The head contains a single row, so you do not need to worry about the **row-loop**. The following example shows a full head renderer (with sorting).
```
Comment (thayannevls) [inline react/components/EXPERIMENTAL_Table/docs/Advanced.md]: This text is the same as the previous section :thinking:

File: react/components/EXPERIMENTAL_Table/Sections/Cell.tsx [hunk 2]
```diff
@@ -1,53 +1,82 @@
...
 import CaretUp from '../../icon/CaretUp/index.js'
-
-const SUFIX_GAP = 0.5
+import { ComposableWithRef } from '../types'
 
 const HoverContext = createContext<boolean>(false)
 
-function HoverProvider({ children, value }) {
-  return <HoverContext.Provider value={value}>{children}</HoverContext.Provider>
+export enum CellTag {
```
Comment (thayannevls) [inline react/components/EXPERIMENTAL_Table/Sections/Cell.tsx]: There is a discussion about using `const enum` instead of `enum`, not sure if it applies in this case but I would check it out

File: react/components/EXPERIMENTAL_Table/Sections/index.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+import { useTestingContext } from '../context/testing'
+import { useLoadingContext } from '../context/loading'
+import Tbody, { ComposableTbody } from './Tbody'
+import Thead, { ComposableThead } from './Thead'
+
+type Props = PropsWithChildren<E2ETestable & HasMotion & NativeTable>
+
+function Sections(
+  { children, className, motion }: Props,
+  ref: Ref<HTMLTableElement>
```
Comment (thayannevls) [inline react/components/EXPERIMENTAL_Table/Sections/index.tsx]:
```suggestion
  ref: Ref<HTMLTableElement>,
```
Eslint didn't throw an error about this? :thinking:

File: react/components/EXPERIMENTAL_Table/context/data.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  columns: Column[]
+  items: Items
+}
+
+const DataContext = createContext<Data>(null)
+
+export function useDataContext() {
+  const context = useContext(DataContext)
+  if (!context) {
+    throw new Error('Do not use Data outside of the Table')
```
Comment (thayannevls) [inline react/components/EXPERIMENTAL_Table/context/data.tsx]: Can't you throw a more specific error? :eyes:

File: react/components/EXPERIMENTAL_Table/hooks/useTableMeasures.tsx [hunk 5]
```diff
@@ -27,15 +28,21 @@ export default function useTableMeasures({
...
-  density: Density
+  density?: Density
+  headless?: boolean
 }
 
 export function calculateTableHeight(
   rowHeight: number,
-  tableSize: number
+  tableSize: number,
+  headless: boolean
```
Comment (thayannevls) [inline react/components/EXPERIMENTAL_Table/hooks/useTableMeasures.tsx]: same thing about lint

File: react/components/EXPERIMENTAL_Table/types.ts [hunk 6]
```diff
@@ -23,3 +32,43 @@ export type Column = {
...
+/** Type with children that receives render-props */
+export type RenderProps<P = {}, T = Record<string, unknown>> = P & {
+  children?: (data: T) => ReactNode
+}
+
+/** Ref foward component with render props 
+export interface RFCRP<T, P = {}, R = Record<string, unknown>> {
+  (props: RenderProps<P, R>, ref: React.Ref<T>): React.ReactElement | null
+}
+*/
```
Comment (rerissondaniel) [inline react/components/EXPERIMENTAL_Table/types.ts]: If its not being used it should be removed

Comment (matheusps) [reply react/components/EXPERIMENTAL_Table/types.ts]: sure. nice catch

File: react/components/EXPERIMENTAL_Table/docs/Advanced.md (hunk 1 above)
Comment (matheusps) [reply react/components/EXPERIMENTAL_Table/docs/Advanced.md]: oops, cmd + c/v 😅. nice catch

File: react/components/EXPERIMENTAL_Table/Sections/Thead.tsx [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+) {
+  const { testId } = useTestingContext()
+  const { sorting, sticky } = useHeadContext()
+  return (
+    <thead
+      ref={ref}
+      data-testid={`${testId}__header`}
+      className={`w-100 ph4 truncate overflow-x-hidden c-muted-2 f6 ${className}`}
+      {...restProps}>
+      <Row height={TABLE_HEADER_HEIGHT} data="">
```
Comment (emersonlaurentino) [inline react/components/EXPERIMENTAL_Table/Sections/Thead.tsx]: why is this `data =" "` necessary?

Comment (matheusps) [reply react/components/EXPERIMENTAL_Table/Sections/Thead.tsx]: actually, shouldn't. fixed.

File: CHANGELOG.md [hunk 8]
```diff
@@ -25,8 +27,6 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [9.112.23] - 2020-03-18
 
-### Fixed
```
Comment (emersonlaurentino) [inline CHANGELOG.md]: why did you remove?

File: CHANGELOG.md [hunk 9]
```diff
@@ -7,6 +7,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+- Expose `EXPERIMENTAL_Table` sections.
```
Comment (emersonlaurentino) [inline CHANGELOG.md]: Should it be ### Added?

File: CHANGELOG.md (hunk 8 above)
Comment (matheusps) [reply CHANGELOG.md]: some rebasing issue, I suppose. fixed!

File: CHANGELOG.md (hunk 9 above)
Comment (matheusps) [reply CHANGELOG.md]: done

File: react/components/EXPERIMENTAL_Table/Sections/Cell.tsx (hunk 2 above)
Comment (matheusps) [reply react/components/EXPERIMENTAL_Table/Sections/Cell.tsx]: I'm keeping the `enum` because they will only be relevant for `graphql` in the future. (also, we need additional plugins on babel for that 😅)
